libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -1569,10 +1569,18 @@ internal object MapboxRouteLineUtils {
     internal fun getAlternativeRouteDeviationOffsets(
         metadata: AlternativeRouteMetadata
     ): Double {
-        return distinctGranularDistancesProvider(metadata.navigationRoute)?.let {
+        return distinctGranularDistancesProvider(metadata.navigationRoute)?.let { distances ->
             val index = metadata.forkIntersectionOfAlternativeRoute.geometryIndexInRoute
-            val distanceRemaining = it.distancesArray[index].distanceRemaining
-            1.0 - distanceRemaining / it.distance
+            val distanceRemaining = distances.distancesArray.getOrElse(index) {
+                logE(
+                    "Remaining distance at index '$it' requested but there are " +
+                        "${distances.distancesArray.size} elements in the distances array. " +
+                        "The alternative with ID '${metadata.navigationRoute.id}' will be hidden.",
+                    LOG_CATEGORY
+                )
+                distances.distancesArray.last()
```

Comment:
This will crash with `NoSuchElementException` if `distances.distancesArray` is empty.

Reply:
That's right, I even had a code that caught that as well. However, this case shouldn't ever happen and because of that I didn't have a way to write an instrumentation test against which made me opt to remove it. If that could actually happen, we'd crash much earlier in the processing pipeline from looking at the code.

I could write an artificial unit test against it but since this all is a workaround, I was on the fence.

Reply:
Reading my own comment again, this isn't the right choice. I'll adjust, thanks for making me think!